prescient.el

```diff
@@ -327,47 +347,65 @@ as a sub-query delimiter."
       (format "\\(%s\\)" regexp)
     regexp))
 
-(defun prescient-literal-regexp (query &optional with-groups)
+(defun prescient-literal-regexp (query props)
   "Return a regexp matching QUERY with character folding.
-If WITH-GROUPS is `all', enclose the match in a capture group."
+If property `:with-groups' in PROPS is `all', enclose the match
```

Comment:
After using `cl-defun`, `with-groups` becomes a normal argument from the point of view of checkdoc.

```suggestion
If WITH-GROUPS is `all', enclose the match
```